dmn-engine/src/main/scala/org/camunda/dmn/evaluation/ContextEvaluator.scala

```diff
@@ -1,28 +1,28 @@
 package org.camunda.dmn.evaluation
 
-import scala.collection.JavaConverters._
-
 import org.camunda.dmn.DmnEngine._
 import org.camunda.dmn.FunctionalHelper._
-import org.camunda.feel.FeelEngine
 import org.camunda.feel.context.Context._
-import org.camunda.bpm.model.dmn.instance.{Context, ContextEntry, Expression}
 import org.camunda.dmn.parser.{ParsedContext, ParsedDecisionLogic}
-import org.camunda.feel.syntaxtree.{Val, ValContext, ValFunction}
+import org.camunda.feel.syntaxtree.{Val, ValContext, ValError, ValFunction}
 import org.camunda.dmn.Audit.ContextEvaluationResult
 
 class ContextEvaluator(
     eval: (ParsedDecisionLogic, EvalContext) => Either[Failure, Val]) {
 
   def eval(context: ParsedContext, ctx: EvalContext): Either[Failure, Val] = {
 
-    evalContextEntries(context.entries, ctx).right.flatMap(results =>
-      evalContextResult(context.aggregationEntry, results, ctx).right.map {
-        result =>
+    evalContextEntries(context.entries, ctx).flatMap(results =>
+      evalContextResult(context.aggregationEntry, results, ctx) match {
+        case r @ Right(result) =>
           ctx.audit(context,
                     ContextEvaluationResult(entries = results, result = result))
-
-          result
+          r
+        case l @ Left(failure) =>
+          ctx.audit(context,
+                    ContextEvaluationResult(entries = results,
+                                            result = ValError(failure.message)))
+          l
```

Comment:
We see this pattern in different places. For readability and maintenance reasons, we should implement this pattern only once.

For example, by introducing a new method in the audit that can handle `Either[Failure, Val]`. Maybe something like this: 
```
def asEvaluationResult(result: Either[Failure, Val], f: Val => EvaluationResult) = {
      result match {
        case Right(r) => f(r)
        case Left(failure) => f(ValError(failure.message))
      }
    }
```

Reply:
How about we add an audit function that does both cases. See https://github.com/camunda/dmn-scala/pull/16/commits/d5b2c368fecdde2de6d80919415da24d49be409b

Reply:
I like the idea. I added suggestions to improve the readability.